### PR TITLE
feat: Languages page

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds the `/languages` page

## Description

- adds `languages.tsx` React page
- adds `translations` utils
- adds `Language` interface

## Preview

https://deploy-preview-150--ethereum-org-fork.netlify.app/languages

### CODE REVIEW

File: src/pages/languages.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import OldHeading from "../components/OldHeading"
+
+import { getRequiredNamespacesForPath } from "@/lib/utils/translations"
+import { languageMetadata } from "@/lib/utils/languages"
+import { getLastDeployDate } from "@/lib/utils/getLastDeployDate"
+
+import { TranslationKey } from "@/lib/types"
+import { Language } from "@/lib/interfaces"
+
+export const getStaticProps: GetStaticProps = async (context) => {
```
Comment (pettinarip) [inline src/pages/languages.tsx]: Lets use the `satisfies` form as we are doing in [the `slug` file](https://github.com/ethereum/ethereum-org-fork/blob/dev/src/pages/%5B...slug%5D.tsx#L171) for consistency and improved type checking.

```suggestion
export const getStaticProps = async (context) => {
```

Comment (pettinarip) [reply src/pages/languages.tsx]: We could also declare the props being used here
```ts
type Props = SSRConfig & {
  lastDeployDate: string
}

// and then
... satisfies GetStaticProps<Props>
```

File: src/lib/utils/translations.ts [hunk 2]
```diff
@@ -18,21 +18,35 @@ export const getLocaleForNumberFormat = (locale: Lang): Lang =>
...
   return i18nConfigs.map((language) => language.code).includes(lang)
-} 
+}
+
+export const getRequiredNamespacesForPath = (
+  path: string,
+  layout?: string | undefined
+) => {
+  // For pages that don't need extra namespaces or don't use the Translation component
+  // if (!path) return "common"
```
Comment (pettinarip) [inline src/lib/utils/translations.ts]: Is this something you would want to elaborate on? otherwise, can we remove the comment?

File: src/lib/utils/languages.ts
Comment (pettinarip) [inline src/lib/utils/languages.ts:1]: Is this file needed? seems redundant having https://github.com/ethereum/ethereum-org-fork/blob/dev/src/lib/utils/translations.ts where most of these functions are repeated.

File: src/lib/interfaces.ts [hunk 3]
```diff
@@ -163,3 +169,13 @@ export interface Author {
     url: string
   }
 }
+
+// Locales
+export interface Language {
```
Comment (pettinarip) [inline src/lib/interfaces.ts]: This type is now called `I18nLocale` and lives in the `types.ts` file.

**Review by pettinarip — CHANGES_REQUESTED** [review]

File: src/lib/utils/translations.ts (hunk 2 above)
Comment (nhsz) [reply src/lib/utils/translations.ts]: Removed

File: src/lib/utils/languages.ts
Comment (nhsz) [reply src/lib/utils/languages.ts:1]: Removed and moved utils to `translations.ts`

File: src/pages/languages.tsx (hunk 1 above)
Comment (nhsz) [reply src/pages/languages.tsx]: Fixed

File: src/lib/utils/translations.ts [hunk 4]
```diff
@@ -1,9 +1,45 @@
...
+  return { ...result, [config.code]: config }
+}, {} as Languages)
+
+const buildLangs = (process.env.GATSBY_BUILD_LANGS || "")
+  .split(",")
+  .filter(Boolean)
+
+// will take the same shape as `languages`. Only thing we are doing
+// here is filtering the desired langs to be built
+export const languageMetadata: {
```
Comment (pettinarip) [inline src/lib/utils/translations.ts]: @nhsz I see that we are just importing the `languageMetadata` in the languages page.

Most of the things defined here are not used anymore since those things are handled now in the `next-i18next.config.js` file.

Or am I missing something?

I think we could get rid of: 
- `buildLangs`
- `languageMetadata` (and just export and use the `languages` object)
- `supportedLanguages`
- `ignoreLanguages`

LMK what do you think

Comment (nhsz) [reply src/lib/utils/translations.ts]: Fixed

**Review by pettinarip — APPROVED** [review]
LGTM! nice job @nhsz
